### PR TITLE
Add --with-product-suffix configure option

### PR DESCRIPTION
**Pull request by AdamBrousseau**
Signed-off-by: Adam Brousseau <adam.brousseau@ca.ibm.com>

### CODE REVIEW

**Comment by AdamBrousseau** [comment]
https://openj9-jenkins.osuosl.org/job/Build_JDK11_s390x_linux_Personal/136/
```
--with-product-name="Adam" --with-product-suffix="Brousseau"
...
23:31:42  + build/linux-s390x-normal-server-release/images/jdk/bin/java -Xjit -version
23:31:43  openjdk version "11.0.12-internal" 2021-07-20
23:31:43  Adam Brousseau (build 11.0.12-internal+0-adhoc.jenkins.BuildJDK11s390xlinuxPersonal)
23:31:43  Eclipse OpenJ9 VM (build HEAD-da2aa50f03e, JRE 11 Linux s390x-64-Bit Compressed References 20210722_136 (JIT enabled, AOT enabled)
23:31:43  OpenJ9   - da2aa50f03e
23:31:43  OMR      - 88427d6420f
23:31:43  JCL      - 569c79316b0 based on jdk-11.0.12+7)
```

File: make/autoconf/jdk-version.m4 [hunk 1]
```diff
@@ -135,6 +135,19 @@ AC_DEFUN_ONCE([JDKVER_SETUP_JDK_VERSION_NUMBERS],
     PRODUCT_NAME="$with_product_name"
   fi
 
+  # The product suffix, if any
+  AC_ARG_WITH(product-suffix, [AS_HELP_STRING([--with-product-suffix],
+      [Set product suffix. Among other uses, appends the 'java.runtime.name' system property. @<:@not specified@:>@])])
```
Comment (keithc-ca) [inline make/autoconf/jdk-version.m4:140]: The help doesn't read well; suggest replacing 'appends' with 'contributes to'.

Comment (AdamBrousseau) [reply make/autoconf/jdk-version.m4]: updated

**Comment by AdamBrousseau** [comment]
@JasonFengJ9 can you review/merge these 3 PRs. I will prepare similar ones for 17/18 and also the service branches.

**Review by keithc-ca — APPROVED** [review]

**Pull request merged**